pom.xml

```diff
@@ -12,16 +12,38 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
```

Comment:
I will revert this, mainly because it does not really belong to the core of this PR.
Having written that, note that the PR now uses the latest versions of some packages and I am not really sure whether those packages still support 1.8 or require a newer Java version.